### ISSUE TITLE
Bad local variable type, at com.google.android.gms.location.LocationServices.getFusedLocationProviderClient

### ISSUE THREAD

**Issue by thecb4** [robolectric/robolectric#7380]
### Description
Attempting to test Location Services with Robolectric and the FusedLocation Service with Google Play Services. Unclear what I am missing based on the documentation.

### Steps to Reproduce
Source code below

### Robolectric & Android Version
android compile sdk = 31
android target sdk = 31
kotlin version = 1.5.31
jp compose version = 1.1.0-beta01
robolectric 4.8.1
robolectric shadow support 3.3.2
robolectric shadow play services 3.3.2

### Link to a public git repo demonstrating the problem:

```
/**
 * WeatherWear UI Test, using Robolectric.
 *
 * See [testing documentation](http://d.android.com/tools/testing).
 */
@RunWith(AndroidJUnit4::class)
@Config(instrumentedPackages = ["androidx.loader.content"])
class WeatherWearUITest {


  @get:Rule val composeTestRule = createComposeRule()
  @get:Rule val activityTestRule = activityScenarioRule<MainActivity>()

  @Before
  @Throws(Exception::class)
  fun setUp() {
    ShadowLog.stream = System.out // Redirect Logcat to console
  }

  @Test
  fun `when I test, then it should work`() {

    activityTestRule.scenario.onActivity {
      composeTestRule.onNodeWithTag("LocationIdentifier")
        .assertIsDisplayed()
    }

  }

}
```

output:
```
Bad local variable type
Exception Details:
  Location:
    com/google/android/gms/common/api/GoogleApi.<init>(Landroid/content/Context;Lcom/google/android/gms/common/api/Api;Lcom/google/android/gms/common/api/Api$ApiOptions;Lcom/google/android/gms/common/api/internal/StatusExceptionMapper;)V @4: aload
  Reason:
    Type top (current frame, locals[5]) is not assignable to reference type
  Current Frame:
    bci: @4
    flags: { flagThisUninit }
    locals: { uninitializedThis, 'android/content/Context', 'com/google/android/gms/common/api/Api', 'com/google/android/gms/common/api/Api$ApiOptions', 'com/google/android/gms/common/api/internal/StatusExceptionMapper' }
    stack: { uninitializedThis, 'android/content/Context', 'com/google/android/gms/common/api/Api', 'com/google/android/gms/common/api/Api$ApiOptions' }
  Bytecode:
    0000000: 2a2b 2c2d 1905 b602 70b7 0280 2ab6 0267
    0000010: 2a2b 2c2d 1904 ba02 8400 00b1          

java.lang.VerifyError: Bad local variable type
Exception Details:
  Location:
    com/google/android/gms/common/api/GoogleApi.<init>(Landroid/content/Context;Lcom/google/android/gms/common/api/Api;Lcom/google/android/gms/common/api/Api$ApiOptions;Lcom/google/android/gms/common/api/internal/StatusExceptionMapper;)V @4: aload
  Reason:
    Type top (current frame, locals[5]) is not assignable to reference type
  Current Frame:
    bci: @4
    flags: { flagThisUninit }
    locals: { uninitializedThis, 'android/content/Context', 'com/google/android/gms/common/api/Api', 'com/google/android/gms/common/api/Api$ApiOptions', 'com/google/android/gms/common/api/internal/StatusExceptionMapper' }
    stack: { uninitializedThis, 'android/content/Context', 'com/google/android/gms/common/api/Api', 'com/google/android/gms/common/api/Api$ApiOptions' }
  Bytecode:
    0000000: 2a2b 2c2d 1905 b602 70b7 0280 2ab6 0267
    0000010: 2a2b 2c2d 1904 ba02 8400 00b1          

	at com.google.android.gms.location.LocationServices.getFusedLocationProviderClient(com.google.android.gms:play-services-location@@18.0.0:1)
	at io.thecb4.weatherwear.services.location.LocationService.<init>(LocationService.kt:32)
```

**Comment by hoisie** [comment]
Seems to be related to https://github.com/robolectric/robolectric/issues/7269

TLDR is GMS core is shipping jar files to maven that have been desugared, which is a no-no, but I'm not sure we can do anything about that except not insrument their jar.

**Comment by utzcoz** [comment]
Hi @thecb4 , looks like this issue is related to https://github.com/robolectric/robolectric/issues/7269 as @hoisie said at https://github.com/robolectric/robolectric/issues/7380#issuecomment-1153527608. You can pose your questions at that issue, and feel free to reopen this issue if you think they are not related. Thanks.

**Comment by hoisie** [comment]
@thecb4 where is this GoogleApi class from? What package/version?

**Comment by swan-amazon** [comment]
This issue is resolved by #7650 , which was released into [robolectric-4.9](https://github.com/robolectric/robolectric/releases/tag/robolectric-4.9). Please close this issue.
